NCU-Project/Dogs/DogsViewController.swift

```diff
@@ -30,16 +30,19 @@ final class DogsViewController: UIViewController {
         
         tableView.dataSource = nil // Required as dataSource is being replaced
         
-        let input = DogsViewModel.Input(filterString: searchBar.rx.text.asObservable())
+        let searchBarText = searchBar.rx.text.asObservable()
         
+        let input = DogsViewModel.Input(query: searchBarText)
         let output = viewModel.transform(input)
         
-        output.networkRequestResult
+        viewModel.filterDogs(query: searchBarText, dogs: output.networkRequestResult)
             .bind(to: tableView.rx.items(cellIdentifier: "dogCell", cellType: UITableViewCell.self)) { row, dog, cell in
                 cell.textLabel?.text = dog.breed
-            }
-            .disposed(by: disposeBag)
+        }.disposed(by: disposeBag)
         
-        // TODO: Bind searchBar to data
+        searchBar.rx.searchButtonClicked.asObservable()
+            .bind {
+                self.searchBar.resignFirstResponder()
```

Comment:
nice one, good choice of keeping this in view controller
can add `guard let strongSelf = self else { return }` because of possible memory leak here